test/built-ins/Temporal/PlainYearMonth/prototype/with/calendar-arguments.js

```diff
@@ -12,15 +12,18 @@ includes: [temporalHelpers.js]
 features: [Temporal]
 ---*/
 
-const options = {};
+const options = {
+  extra: "property",
+};
 class CustomCalendar extends Temporal.Calendar {
   constructor() {
     super("iso8601");
   }
   yearMonthFromFields(...args) {
     assert.sameValue(args.length, 2, "args.length");
     assert.sameValue(typeof args[0], "object", "args[0]");
-    assert.sameValue(args[1], options, "args[1]");
+    assert.notSameValue(args[1], options, "args[1] is a copy of options");
+    assert.sameValue(args[1].extra, "property", "All properties are copied");
```

Comment:
Consider whether you want to verify the prototype for these

Reply:
Added, and found that ZonedDateTime didn't have a test like this one yet, so added that as well.